diff --git i/setup.py w/setup.py
index 5bb6e215..3015c07f 100644
--- i/setup.py
+++ w/setup.py
@@ -29,7 +29,6 @@ if sys.version_info < (3, 10):
 import setuptools
 
 import atexit
-import distutils.command.build
 import os
 import shlex
 import shutil
@@ -79,14 +78,14 @@ class SdistCommand(setuptools.command.sdist.sdist):
     super().make_release_tree(base_dir, files)
 
 
-class BuildCommand(distutils.command.build.build):
+class BuildCommand(setuptools.command.build_py):
 
   def finalize_options(self):
-    if self.build_base == 'build':
+    if self.build_lib == 'build':
       # Use temporary directory instead, to avoid littering the source directory
       # with a `build` sub-directory.
       tempdir = tempfile.TemporaryDirectory()
-      self.build_base = tempdir.name
+      self.build_lib = tempdir.name
       atexit.register(tempdir.cleanup)
     super().finalize_options()
 
