--- spack.src/configure   2025-02-21 12:14:19.000000000 +0100
+++ spack.src.patched/configure   2025-02-21 12:15:38.000000000 +0100
@@ -2629,10 +2629,9 @@
   if ($config_tk) { $tcl_libs = "-ltk8.5 -lX11 " . $tcl_libs; }
   $vmd_libs          = "$tcl_libs -lz";

-  $arch_nvcc     = "/usr/local/cuda/bin/nvcc";
+  $arch_nvcc     = "nvcc";
   $arch_nvccflags   = "--ptxas-options=-v " .
-                    "-gencode arch=compute_30,code=compute_30 " .
-                    "-gencode arch=compute_70,code=compute_70 " .
+                    "-gencode arch=compute_90,code=compute_90 " .
                     "--ftz=true ";
   $cuda_library     = "-L/usr/local/cuda/lib64";
   $arch_cc       = "cc";
