diff --git i/aten/src/ATen/native/cpu/Unfold2d.cpp w/aten/src/ATen/native/cpu/Unfold2d.cpp
index 8c94decfff0..f89181462f8 100644
--- i/aten/src/ATen/native/cpu/Unfold2d.cpp
+++ w/aten/src/ATen/native/cpu/Unfold2d.cpp
@@ -169,7 +169,7 @@ static void unfolded2d_acc_channels_last(
 
 /* note: due to write issues, this one cannot be parallelized as well as
  * unfolded2d_copy */
-#if defined(__GNUC__) && __GNUC__ == 14 && defined(__ARM_FEATURE_SVE) && !defined(__ARM_FEATURE_BF16)
+#if defined(__GNUC__) && __GNUC__ == 14 && defined(__ARM_FEATURE_SVE)
 // Workaround for gcc-14.2.0 ICE during RTL pass: vregs when compiling for SVE without BF16
 __attribute__((optimize("no-tree-vectorize")))
 #endif
