diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1334bb6da5..028c9ab03a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1647,6 +1647,7 @@ else()
     add_library(cp2k STATIC "${CP2K_SRCS};${CP2K_SRCS_C};${CP2K_SRCS_GPU}")
   endif()
 endif()
+set_target_properties(cp2k PROPERTIES LINKER_LANGUAGE CXX)
 
 target_compile_features(cp2k PUBLIC cuda_std_11)
 target_include_directories(cp2k PUBLIC $<INSTALL_INTERFACE:include/cp2k>)
