diff --git a/.clang-tidy b/.clang-tidy
index 40374e3..6a6ab5e 100644
--- a/.clang-tidy
+++ b/.clang-tidy
@@ -21,7 +21,10 @@ Checks: >
   -readability-uppercase-literal-suffix,
   -readability-avoid-unconditional-preprocessor-if,
   -readability-use-anyofallof,
-  -readability-function-cognitive-complexity
+  -readability-function-cognitive-complexity,
+  -portability-avoid-pragma-once,
+  -modernize-use-scoped-lock,
+  -readability-use-concise-preprocessor-directives
 
 WarningsAsErrors: "*"
 HeaderFileExtensions: ['h','hpp']
