From 7ba2c4f82fcbbde6586607f0fe0bd01b867fdb9e Mon Sep 17 00:00:00 2001
From: Simon Pintarelli <simon.pintarelli@cscs.ch>
Date: Mon, 2 Feb 2026 23:40:39 +0100
Subject: [PATCH] patch sockets.c

---
 Modules/sockets.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Modules/sockets.c b/Modules/sockets.c
index 50deffa..40b694f 100644
--- a/Modules/sockets.c
+++ b/Modules/sockets.c
@@ -43,6 +43,7 @@ Functions:
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <sys/un.h>
+#define __USE_GNU
 #include <netdb.h>
 
 void open_socket(int *psockfd, int* inet, int* port, const char* host)
-- 
2.52.0

