diff --git a/cmake/detect-features-c.cmake b/cmake/detect-features-c.cmake
index 313c407c1..3f867b067 100644
--- a/cmake/detect-features-c.cmake
+++ b/cmake/detect-features-c.cmake
@@ -370,12 +370,7 @@ int main() {
 if(${NETWORK} STREQUAL "ofi" OR ${NETWORK} STREQUAL "ofi-crayshasta" OR ${NETWORK} STREQUAL "ofi-linux")
 # assume HPC installation
 include(CMakePrintHelpers)
-  find_package(EnvModules REQUIRED)
   find_package(PkgConfig REQUIRED)
-  if(EnvModules_FOUND)
-  #at least get libfabric loaded if it isn't already
-	env_module(load libfabric)
-  endif()
   set(tmp ${CMAKE_REQUIRED_LIBRARIES})
   if(${PkgConfig_FOUND})
 # this is tortured because pkg-config and cmake are infuriating
diff --git a/cmake/detect-features-cxx.cmake b/cmake/detect-features-cxx.cmake
index 64d4067cb..bd627f14d 100644
--- a/cmake/detect-features-cxx.cmake
+++ b/cmake/detect-features-cxx.cmake
@@ -168,14 +168,7 @@ set(CMAKE_REQUIRED_FLAGS "")
 if(${NETWORK} STREQUAL "ofi" OR ${NETWORK} STREQUAL "ofi-crayshasta" OR ${NETWORK} STREQUAL "ofi-linux")
 # assume HPC installation with LMOD
   include(CMakePrintHelpers)
-  find_package(EnvModules REQUIRED)
   find_package(PkgConfig REQUIRED)
-  if(EnvModules_FOUND)
-  # we need libfabric cray-libpals and cray-pmi
-	env_module(load libfabric)
-	env_module(load cray-libpals)
-	env_module(load cray-pmi)
-  endif()
   set(tmp ${CMAKE_REQUIRED_LIBRARIES})
   if(${PkgConfig_FOUND})
 # this is tortured because pkg-config and cmake are infuriating
diff --git a/src/arch/ofi-crayshasta/conv-mach.h b/src/arch/ofi-crayshasta/conv-mach.h
index bef80a949..b66673fe1 100644
--- a/src/arch/ofi-crayshasta/conv-mach.h
+++ b/src/arch/ofi-crayshasta/conv-mach.h
@@ -74,7 +74,7 @@
 #define CMK_LBDB_ON					   1
 
 #define CMK_64BIT                      1
-#define CMK_AMD64                      1
+#define CMK_ARM                        1
 #ifdef CMK_HAS_GET_MYADDRESS
 #undef CMK_HAS_GET_MYADDRESS
 #define CMK_HAS_GET_MYADDRESS 0
