diff --git a/cmake/costaConfig.cmake.in b/cmake/costaConfig.cmake.in
index 8fae534..d744c90 100644
--- a/cmake/costaConfig.cmake.in
+++ b/cmake/costaConfig.cmake.in
@@ -10,7 +10,7 @@ include(CMakeFindDependencyMacro)
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR} ${CMAKE_MODULE_PATH})
 
 set(MKL_ROOT "@MKL_ROOT@")
-set(MPI_CXX_COMPILER "@MPI_CXX_COMPILER@")
+#set(MPI_CXX_COMPILER "@MPI_CXX_COMPILER@")
 
 set(MPI_DETERMINE_LIBRARY_VERSION TRUE)
 set(COSTA_SCALAPACK "@COSTA_SCALAPACK@")
