diff --git a/prov/lnx/src/lnx_ops.c b/prov/lnx/src/lnx_ops.c
index ba3f097..2d6c187 100644
--- a/prov/lnx/src/lnx_ops.c
+++ b/prov/lnx/src/lnx_ops.c
@@ -455,6 +455,7 @@ ssize_t lnx_trecv(struct fid_ep *ep, void *buf, size_t len, void *desc,
 	struct lnx_ep *lep;
 	const struct iovec iov = {.iov_base = buf, .iov_len = len};
 
+	cuda_set_sync_memops(buf);
 	lep = container_of(ep, struct lnx_ep, le_ep.ep_fid.fid);
 	if (!lep)
 		return -FI_ENOSYS;
@@ -666,6 +667,7 @@ ssize_t lnx_tsenddata(struct fid_ep *ep, const void *buf, size_t len, void *desc
 	fi_addr_t core_addr;
 	void *core_desc = desc;
 
+	cuda_set_sync_memops(buf);
 	lep = container_of(ep, struct lnx_ep, le_ep.ep_fid.fid);
 	if (!lep)
 		return -FI_ENOSYS;
