--- ascent.ori/src/cmake/thirdparty/SetupPython.cmake	2025-02-27 09:25:21.000000000 +0100
+++ ascent/src/cmake/thirdparty/SetupPython.cmake	2025-02-27 09:56:00.000000000 +0100
@@ -47,11 +47,12 @@
         #       We may need several queries and a list of paths to replace
         #       get_python_lib()
         #
-        execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c"
-                                "import sys;from distutils.sysconfig import get_python_lib;sys.stdout.write(get_python_lib())"
-                        OUTPUT_VARIABLE PYTHON_SITE_PACKAGES_DIR
-                        ERROR_VARIABLE ERROR_FINDING_SITE_PACKAGES_DIR)
-        MESSAGE(STATUS "PYTHON_SITE_PACKAGES_DIR ${PYTHON_SITE_PACKAGES_DIR}")
+        #execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c"
+        #                        "import sys;import os;from distutils.sysconfig import get_python_lib;sys.stdout.write(f'{os.path.dirname(get_python_lib())}/site-packages')"
+        #                OUTPUT_VARIABLE PYTHON_SITE_PACKAGES_DIR
+        #                ERROR_VARIABLE ERROR_FINDING_SITE_PACKAGES_DIR)
+        set(PYTHON_SITE_PACKAGES_DIR "/user-environment/env/develop/lib/python3.11/site-packages")
+        MESSAGE(STATUS "PYTHON_SITE_PACKAGES_DIR === ${PYTHON_SITE_PACKAGES_DIR}")
 
         if(NOT EXISTS ${PYTHON_SITE_PACKAGES_DIR})
             MESSAGE(FATAL_ERROR "Reported PYTHON_SITE_PACKAGES_DIR ${PYTHON_SITE_PACKAGES_DIR} does not exist!")
