diff --git a/projects/rocprofiler-compute/src/rocprof_compute_profile/profiler_rocprofiler_sdk.py b/projects/rocprofiler-compute/src/rocprof_compute_profile/profiler_rocprofiler_sdk.py
index aa29be9e8c..0e3f4d042c 100644
--- a/projects/rocprofiler-compute/src/rocprof_compute_profile/profiler_rocprofiler_sdk.py
+++ b/projects/rocprofiler-compute/src/rocprof_compute_profile/profiler_rocprofiler_sdk.py
@@ -66,6 +66,10 @@ class rocprofiler_sdk_profiler(RocProfCompute_Base):
             args.rocprofiler_sdk_library_path,
             rocprofiler_attach_tool_path,
         ]
+        # Add any paths already in LD_PRELOAD variable
+        existing_ld_preload = os.environ.get("LD_PRELOAD", "")
+        if existing_ld_preload:
+            ld_preload.extend(existing_ld_preload.split(":"))
         options = {
             "ROCPROFILER_LIBRARY_CTOR": "1",
             "LD_PRELOAD": ":".join(ld_preload),
diff --git a/projects/rocprofiler-compute/src/utils/utils.py b/projects/rocprofiler-compute/src/utils/utils.py
index 2ae5020d7c..da6becbf15 100644
--- a/projects/rocprofiler-compute/src/utils/utils.py
+++ b/projects/rocprofiler-compute/src/utils/utils.py
@@ -962,6 +962,10 @@ def pc_sampling_prof(
             rocprofiler_sdk_tool_path,
             rocprofiler_sdk_library_path,
         ]
+        # Add any paths already in LD_PRELOAD variable
+        existing_ld_preload = os.environ.get("LD_PRELOAD", "")
+        if existing_ld_preload:
+            ld_preload.extend(existing_ld_preload.split(":"))
         options = {
             "ROCPROFILER_LIBRARY_CTOR": "1",
             "LD_PRELOAD": ":".join(ld_preload),
