diff -Naur scalasca-2.6.2.ori/build-config/common/platforms/platform-backend-craygeneric scalasca-2.6.2/build-config/common/platforms/platform-backend-craygeneric
--- scalasca-2.6.2.ori/build-config/common/platforms/platform-backend-craygeneric	2025-04-02 08:37:55.000000000 +0200
+++ scalasca-2.6.2/build-config/common/platforms/platform-backend-craygeneric	2025-04-11 15:56:20.000000000 +0200
@@ -1,7 +1,7 @@
-CC=cc
-CXX=CC
-F77=ftn
-FC=ftn
+CC=mpicc
+CXX=mpicxx
+F77=mpif90
+FC=mpif90
 
 #CPPFLAGS=
 CFLAGS=-g -O2
diff -Naur scalasca-2.6.2.ori/build-config/common/platforms/platform-mpi-craygeneric scalasca-2.6.2/build-config/common/platforms/platform-mpi-craygeneric
--- scalasca-2.6.2.ori/build-config/common/platforms/platform-mpi-craygeneric	2025-04-02 08:37:55.000000000 +0200
+++ scalasca-2.6.2/build-config/common/platforms/platform-mpi-craygeneric	2025-04-11 15:56:45.000000000 +0200
@@ -1,7 +1,7 @@
-MPICC=cc
-MPICXX=CC
-MPIF77=ftn
-MPIFC=ftn
+MPICC=mpicc
+MPICXX=mpicxx
+MPIF77=mpif90
+MPIFC=mpif90
 
 #MPI_CPPFLAGS=
 MPI_CFLAGS=-g -O2
diff -Naur scalasca-2.6.2.ori/build-config/common/platforms/platform-shmem-craygeneric scalasca-2.6.2/build-config/common/platforms/platform-shmem-craygeneric
--- scalasca-2.6.2.ori/build-config/common/platforms/platform-shmem-craygeneric	2025-04-02 08:37:55.000000000 +0200
+++ scalasca-2.6.2/build-config/common/platforms/platform-shmem-craygeneric	2025-04-11 15:57:05.000000000 +0200
@@ -1,7 +1,7 @@
-SHMEMCC=cc
-SHMEMCXX=CC
-SHMEMF77=ftn
-SHMEMFC=ftn
+SHMEMCC=mpicc
+SHMEMCXX=mpicxx
+SHMEMF77=mpif90
+SHMEMFC=mpif90
 
 #SHMEM_CPPFLAGS=
 SHMEM_CFLAGS=-g -O2
